### PR TITLE
PFC/PFCWD/ECN tests 

### PR DESCRIPTION
**Pull request by ANISH-GOTTAPU**
### Description of PR
Migration of PFC/PFCWD/ECN tests from `abstract open traffic generator` to `snappi`

Summary:
Test case(new/improvement)

### Type of change
- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x ] Test case(new/improvement)

#### What is the motivation for this PR?
The motivation for this PR is to migrate the tests that are written with `abstract open traffic generator` to `snappi`

#### How did you do it?
Replaced the code where abstract is used to snappi

#### How did you verify/test it?
verified all the tests in sonic-mgmt using pytest with DUT and IXIA connections 
snappi version used for the development of tests `snappi[ixnetwork]==0.3.13`

### CODE REVIEW

**Comment by ANISH-GOTTAPU** [comment]
adding snappi[ixnetwork] installation in Dockerfile.j2 will be addressed in this https://github.com/Azure/sonic-buildimage/pull/7204, The version will be upgraded to snappi[ixnetwork]==0.3.13

File: tests/common/ixia/ixia_fixtures.py [hunk 1]
```diff
@@ -193,11 +195,34 @@ def ixia_api(ixia_api_serv_ip,
     if api_session:
         api_session.assistant.Session.remove()
 
+
+@pytest.fixture(scope='module')
+def snappi_api(ixia_api_serv_ip,
```
Comment (neethajohn) [inline tests/common/ixia/ixia_fixtures.py:200]: please move the snappi api's to a diff file

File: tests/common/ixia/ixia_fixtures.py [hunk 2]
```diff
@@ -224,17 +249,13 @@ def ixia_testbed(conn_graph_facts,
 
     ixia_ports = ixia_fanout_list.get_ports(peer_device=duthost.hostname)
 
-    ports = list()
-    port_names = list()
     port_speed = None
 
     """ L1 config """
+    config = snappi_api.config()
```
Comment (neethajohn) [inline tests/common/ixia/ixia_fixtures.py:255]: let's leave this method intact and have the snappi setup done in a separate method

**Review by neethajohn — CHANGES_REQUESTED** [review]

**Comment by neethajohn** [comment]
Please split this PR into smaller ones - a separate one for pfc, pfcwd and ecn which will make it easier to review

**Comment by ANISH-GOTTAPU** [comment]
closing this PR as per review comments, opened https://github.com/Azure/sonic-mgmt/pull/3298 for PFC to address review comments

**Pull request closed without merging**